### PR TITLE
Recalculate on window resize and <details> toggle

### PR DESCRIPTION
**Pull request by Aprillion**

### CODE REVIEW

File: src/index.jsx [hunk 1]
```diff
@@ -115,6 +115,27 @@ function AutoHeight({children, element = 'div', className: propClassName = '', .
 
   useEffect(updateHeight)
 
+  // recalculate height on <details> toggle and window resize by forcing rerender
+  const [_, setStateToRerender] = useState()
+  useEffect(() => {
+    let isMounted = true
```
Comment (mruwnik) [inline src/index.jsx:121]: is this in case the eventlistners don't get destroyed properly? Or more to handle possible race conditions?

**Review by mruwnik — APPROVED** [review]
Am I correct in thinking that this sets the state just to trigger the rerender?

File: src/index.jsx (hunk 1 above)
Comment (Aprillion) [reply src/index.jsx:121]: yeah, the useless memory leak warning that was removed in React 18 https://github.com/facebook/react/pull/22114 ... but the library supports React 17 too, so let's not scare ourselves when using this library in Stampy 😅

Comment (Aprillion) [reply src/index.jsx:121]: and you are correct, it sets state to rerender .. moving code from https://github.com/StampyAI/stampy-ui/blob/3e3ae3ec4e7f4f1ad5a3ef70c3daf424eab92a3e/app/hooks/useRerenderOnResize.ts - but it needs to be new object every time to work, the `set(empty)` was a bug, so let's fix that here in the library and not in distant user code

**Pull request merged**